wallet/core/src/utils.rs

```diff
@@ -81,7 +97,12 @@ fn str_to_sompi(amount: &str) -> Result<u64> {
     let integer = amount[..dot_idx].parse::<u64>()? * SOMPI_PER_KASPA;
     let decimal = &amount[dot_idx + 1..];
     let decimal_len = decimal.len();
-    let decimal =
-        if decimal_len <= 8 { decimal.parse::<u64>()? * 10u64.pow(8 - decimal_len as u32) } else { decimal[..8].parse::<u64>()? };
+    let decimal = if decimal_len == 0 {
+        0
+    } else if decimal_len <= 8 {
+        decimal.parse::<u64>()? * 10u64.pow(8 - decimal_len as u32)
+    } else {
+        decimal[..8].parse::<u64>()?
```

Comment:
Optional: This condition should never happen in a legitimate KAS value. So it might be worth changing this to an Err since this would mean you were passed a string that had 9+ decimals - which could mean something is off elsewhere in the code that made that happen.

Reply:
Hm..  It's a valid point...  The function is constructed and is used mainly for user input and I am not sure of all the pros or cons of both cases.   Like if you copy/paste a value from somewhere, it can have a very long fractional value...  and I am not sure whether such a value should be rejected or truncated.  It kind of falls into "what is best/friendliest for the user" domain...  technically I think we should perhaps do something like `ceil(v*1e8)/1e8` and then truncate...  this way the user can input anything but will get back a rounded value... but perhaps a separate function should be made for that.

Let me add a TODO to this so that we can take a look at this a bit later on.